packages/twenty-server/src/database/commands/0-22-add-new-address-field-to-views-with-deprecated-address.command.ts

```diff
@@ -0,0 +1,225 @@
+})
+export class AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommand extends CommandRunner {
+  private readonly logger = new Logger(
+    AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommand.name,
+  );
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(BillingSubscription, 'core')
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    @InjectRepository(FeatureFlagEntity, 'core')
+    private readonly featureFlagRepository: Repository<FeatureFlagEntity>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+    private readonly twentyORMManager: TwentyORMManager,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommandOptions,
+  ): Promise<void> {
+    this.logger.log('running');
+    let workspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      workspaceIds = [options.workspaceId];
+    } else {
+      const workspaces = await this.workspaceRepository.find();
+
+      const activeWorkspaceIds = (
+        await Promise.all(
+          workspaces.map(async (workspace) => {
+            const isActive = await this.workspaceIsActive(workspace);
+
+            return { workspace, isActive };
+          }),
+        )
+      )
+        .filter((result) => result.isActive)
+        .map((result) => result.workspace.id);
+
+      workspaceIds = activeWorkspaceIds;
+    }
+
+    if (!workspaceIds.length) {
+      this.logger.log(chalk.yellow('No workspace found'));
+
+      return;
+    } else {
+      this.logger.log(
+        chalk.green(`Running command on ${workspaceIds.length} workspaces`),
+      );
+    }
+
+    for (const workspaceId of workspaceIds) {
+      const viewFieldRepository =
+        await this.twentyORMManager.getRepositoryForWorkspace(
+          workspaceId,
+          ViewFieldWorkspaceEntity,
+        );
+
+      const dataSourceMetadatas =
+        await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
+          workspaceId,
+        );
+
+      for (const dataSourceMetadata of dataSourceMetadatas) {
+        const workspaceDataSource =
+          await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+        if (workspaceDataSource) {
+          try {
+            const newAddressField = await this.fieldMetadataRepository.findBy({
```

Comment:
It feels like you're close to have built a generic command.
Imo (opt1) is to make it fully generic and pass all params as options to the command, (opt2) is to isolate all variables at the very top of the script and make the script generic after that. Maybe (opt2) is good enough for now?

Reply:
yes good point! I will also take note of that on the doc